pulpcore/tests/functional/__init__.py

```diff
@@ -117,6 +117,14 @@ def orphans_cleanup_api_client(pulpcore_client):
     return OrphansCleanupApi(pulpcore_client)
 
 
+@pytest.fixture
+def gen_role(roles_api_client, gen_object_with_cleanup):
+    def _gen_role(**kwargs):
+        return gen_object_with_cleanup(roles_api_client, kwargs)
+
+    yield _gen_role
```

Comment:
```suggestion
    return _gen_role
```

If there isn't cleanup work to do after having it return is more efficient by not having the generator resumed later only to return.